src/SourceLocator/LocatedSourceWithStrippedSourcesDirectory.php

```diff
@@ -10,7 +10,11 @@
 use function strlen;
 use function substr_replace;
 
-/** @internal */
+/**
+ * @internal
+ *
+ * @psalm-immutable
```

Comment:
This object depends on `LocatedSource` being immutable: is it immutable, or are we lying? :thinking:

Reply:
The class only contains getters and boolean checks. Or am I missing something?

Reply:
Is the inner located source also immutable? That's what decides if this one is immutable too :D

Reply:
The base class looks similar. What's wrong here?

Reply:
The base class looks similar. What's wrong there?

Reply:
Just clarifying here: I thought I saw a suppression, but there wasn't one :-) 

Possibly due to older `roave/better-reflection` dependency before your patch :+1: